Use helm upgrade with image digest to update user deployments

alexberndt opened the pull request.

alexberndt commented:
Reverted to draft: need to add the `helm upgrade` portion to other steps in the workflow.

chanange commented on .github/workflows/dagster.yml:
```diff
@@ -123,9 +124,11 @@ jobs:
-          kubectl rollout restart deployment -n dagster dagster-dagster-user-deployments-${{ inputs.image_name }}
-          kubectl rollout status deployment -n dagster dagster-dagster-user-deployments-${{ inputs.image_name }} -w --timeout=10m && echo "Deployment completed" || ( kubectl logs -n dagster -l deployment=${{ inputs.image_name }} && exit 1 )
+          helm upgrade ${{ env.USER_DEPLOYMENT_POD }} dagster-user-deployments --set-string deployments[0].image.tag=${{ needs.run-docker-workflow.outputs.image_digest }}
+          kubectl rollout status deployment -n dagster ${{ env.USER_DEPLOYMENT_POD }} -w --timeout=10m && echo "Deployment completed" || ( kubectl logs -n dagster -l deployment=${{ inputs.image_name }} && exit 1 )
```
This shouldn't be needed anymore
```suggestion
```

alexberndt replied:
Thanks! Do we want to keep the `echo "Deployment completed" || ( kubectl logs -n dagster -l deployment=${{ inputs.image_name }} && exit 1 )` portion?

chanange reviewed (commented):
We also need to update the scheduled build file

alexberndt commented:
The dagster helm chart does not support providing a digest, only a tag (see helm chart values schema: https://artifacthub.io/packages/helm/dagster/dagster-user-deployments?modal=values-schema&path=deployments[0].image.tag

Refactored the PR to use the github SHA instead.

Also, removed the steps to read the image digest in `docker_build_push.yml`
